dtlpymetrics/utils/metrics.py

```diff
@@ -366,13 +366,19 @@ def general_match(matches: Matches,
             first_annotation = [a for a in first_set if a.id == first_annotation_id][0]
             second_annotation = [a for a in second_set if a.id == second_annotation_id][0]
             geometry_score = df.iloc[row_index, col_index]
-            labels_score = Matchers.match_labels(label1=first_annotation.label,
-                                                 label2=second_annotation.label)
-            attribute_score = Matchers.match_attributes(attributes1=first_annotation.attributes,
-                                                        attributes2=second_annotation.attributes)
-
-            # TODO use ignores for final score
-            annotation_score = (geometry_score + attribute_score + labels_score) / 3
+            labels_score = None
+            attribute_score = None
+            scores = list()
+            scores.append(geometry_score)
+            if ignore_attributes is False:
+                attribute_score = Matchers.match_attributes(attributes1=first_annotation.attributes,
+                                                            attributes2=second_annotation.attributes)
+                scores.append(attribute_score)
+            if ignore_labels is False:
+                labels_score = Matchers.match_labels(label1=first_annotation.label,
+                                                     label2=second_annotation.label)
+                scores.append(labels_score)
+            annotation_score = statistics.mean(scores)
```

Comment:
Since using numpy.mean is less safe than statistics.mean. as can be seen here:

![857786175-image](https://github.com/dataloop-ai-apps/dtlpy-metrics/assets/100369377/88286eae-9a29-43e4-a5f2-eff9260af446)